ide/lsp.client/src/org/netbeans/modules/lsp/client/bindings/ErrorProviderBridge.java

```diff
@@ -0,0 +1,108 @@
+ * under the License.
+ */
+package org.netbeans.modules.lsp.client.bindings;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import javax.swing.text.Document;
+import org.netbeans.api.lsp.Diagnostic;
+import org.netbeans.spi.editor.hints.ErrorDescription;
+import org.netbeans.spi.editor.hints.ErrorDescriptionFactory;
+import org.netbeans.spi.editor.hints.HintsController;
+import org.netbeans.spi.editor.hints.Severity;
+import org.netbeans.spi.lsp.ErrorProvider;
+import org.openide.filesystems.FileObject;
+import org.openide.util.RequestProcessor;
+import org.openide.util.WeakListeners;
+
+class ErrorProviderBridge implements Runnable, DocumentListener {
+
+    private final FileObject file;
+    private final RequestProcessor.Task task;
+    private final Collection<? extends ErrorProvider> errorProviders;
+    private final DocumentListener listener;
+
+    ErrorProviderBridge(Document doc, FileObject file, Collection<? extends ErrorProvider> errorProviders, RequestProcessor rp) {
+        this.file = file;
+        this.errorProviders = errorProviders;
+        this.task = rp.create(this);
+        this.listener = WeakListeners.create(DocumentListener.class, this, doc);
+        doc.addDocumentListener(listener);
+    }
+
+    final void start() {
+        task.schedule(0);
+    }
+
+    final void waitFinished() {
+        task.waitFinished();
+    }
+
+    @Override
+    public final void run() {
+        for (ErrorProvider p : errorProviders) {
+            if (p.getClass().getName().equals("org.netbeans.modules.editor.hints.lsp.HintsDiagnosticsProvider")) {
+                continue;
+            }
+            computeHints(ErrorProvider.Kind.ERRORS, p, "lsp:errors");
+            computeHints(ErrorProvider.Kind.HINTS, p, "lsp:hints");
+        }
+    }
+
+    private void computeHints(final ErrorProvider.Kind type, ErrorProvider p, final String prefix) {
+        List<ErrorDescription> arr = new ArrayList<>();
+        ErrorProvider.Context errorCtx = new ErrorProvider.Context(file, type);
+        List<? extends Diagnostic> errors = p.computeErrors(errorCtx);
+        if (errors != null) {
+            for (Diagnostic e : errors) {
+                final Severity s = Severity.ERROR;
+                ErrorDescription descr = ErrorDescriptionFactory.createErrorDescription(s,
+                        e.getDescription(),
+                        file,
+                        e.getStartPosition().getOffset(),
+                        e.getEndPosition().getOffset()
+                );
+                arr.add(descr);
+            }
+            applyHints(prefix, p, arr);
+        }
+    }
+
+    protected void applyHints(final String prefix, ErrorProvider p, List<ErrorDescription> arr) {
+        HintsController.setErrors(file, prefix + ":" + p.getClass().getSimpleName(), arr);
```

Comment:
```suggestion
        HintsController.setErrors(file, prefix + ":" + p.getClass().getName(), arr);
```

To limit ambiguities.